src/components/auth/RegisterPage.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import Form from 'react-validation/build/form';
+import Input from 'react-validation/build/input';
+import Button from 'react-validation/build/button';
+import User from '../../models/User';
+import { register } from '../../state/auth/auth.actionCreators';
+import './RegisterPage.scss';
+
+type Props = {
+    dispatch: Function,
+}
+
+type State = {
+    user: User,
+    submitted: boolean,
+}
+
+class RegisterPage extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {
+        firstName: '',
+        lastName: '',
+        username: '',
+        password: '',
+      },
+      submitted: false,
+    };
+  }
+
+  handleChange = (event) => {
```

Comment:
You can type the event as well! Use `event: SyntheticInputEvent<HTMLInputElement>`